crates/bevy_core_pipeline/src/core_3d/main_transmissive_pass_3d_node.rs

```diff
@@ -52,72 +54,98 @@ impl ViewNode for MainTransmissivePass3dNode {
-                    render_context.command_encoder().copy_texture_to_texture(
-                        target.main_texture().as_image_copy(),
-                        transmission.texture.as_image_copy(),
-                        physical_target_size.to_extents(),
-                    );
-
-                    let mut render_pass =
-                        render_context.begin_tracked_render_pass(render_pass_descriptor.clone());
+
+            let color_attachments = [Some(target.get_color_attachment())];
+            let depth_stencil_attachment = Some(depth.get_attachment(StoreOp::Store));
+
+            render_context.add_command_buffer_generation_task(move |render_device| {
+                // Run the transmissive pass, sorted back-to-front
+                #[cfg(feature = "trace")]
+                let _main_transmissive_pass_3d_span =
+                    info_span!("main_transmissive_pass_3d").entered();
+
+                // Command encoder setup
+                let mut command_encoder =
+                    render_device.create_command_encoder(&CommandEncoderDescriptor {
+                        label: Some("main_transmissive_pass_3d_command_encoder"),
+                    });
+
+                if screen_space_specular_transmission_steps > 0 {
+                    let transmission =
+                        transmission.expect("`ViewTransmissionTexture` should exist at this point");
+
+                    // `transmissive_phase.items` are depth sorted, so we split them into N = `screen_space_specular_transmission_steps`
+                    // ranges, rendering them back-to-front in multiple steps, allowing multiple levels of transparency.
+                    //
+                    // Note: For the sake of simplicity, we currently split items evenly among steps. In the future, we
+                    // might want to use a more sophisticated heuristic (e.g. based on view bounds, or with an exponential
+                    // falloff so that nearby objects have more levels of transparency available to them)
+                    for range in split_range(
+                        0..transmissive_phase.items.len(),
+                        screen_space_specular_transmission_steps,
+                    ) {
+                        // Copy the main texture to the transmission texture, allowing to use the color output of the
+                        // previous step (or of the `Opaque3d` phase, for the first step) as a transmissive color input
+                        command_encoder.copy_texture_to_texture(
+                            target.main_texture().as_image_copy(),
+                            transmission.texture.as_image_copy(),
+                            physical_target_size.to_extents(),
+                        );
+
+                        let render_pass =
+                            command_encoder.begin_render_pass(&RenderPassDescriptor {
```

Comment:
The descriptor should be constructed once before the `if steps > 0`. It's the same in both branches. That would probably help reduce the diff size too.